### PR TITLE
[CST] - Update font awesome icons to va-icons

### PR DESCRIPTION
**Pull request by pmclaren19**
## Summary

- Updated `ClaimsListItem.jsx` so the font awesome `envelope` icon that displays on the ClaimCard is replaced with the va-icon called `mail` 
- Updated `DueDateOld.jsx` so the font awesome `exclamation-triangle` icon that displays on an alert; when the feature flag cstUseClaimDetailsV2 is disabled on the files tab, is replaced with the va-icon called `warning`
- Updated `SubmittedTrackedItem.jsx` so the font awesome `fa-check-circle` icon that displays under Documents Filed; when a document has been reviewed by the VA and the feature flag cstUseClaimDetailsV2 is disabled on the files tab, is replaced with the va-icon called `check_circle`
- Updated `DocumentsFiled.jsx` so the font awesome `fa-check-circle` icon that displays under Documents Filed; when a document has been reviewed by the VA and the feature flag cstUseClaimDetailsV2 is enabled on the files tab, is replaced with the va-icon called `check_circle`
- Updated `ClaimLetterSection.jsx` so the font awesome `chevron right` icon that displays on the main CST page under the 'Your claim letters' header is replaced with the va-icon called `chevron_right`
- Updated `WhatWeAreDoing.jsx` so the font awesome `chevron right` icon that displays on the status tav under the 'What we're doing' header is replaced with the va-icon called `chevron_right`
- Updated `ClaimCardLink.jsx` so the font awesome `chevron right` icon that displays on the cst home page on the claim card, next to the details link is replaced with the va-icon called `chevron_right` 
- Updated OnThisPageLink.jsx so the font awesome `fa-arrow-down` icon that displays on the rated disabilities page under the 'On this page' header is replaced with the va-icon called `arrow_downward`
- Updated `claim-status.scss` so that the active-va-link css occurs on the `va-icon` instead of the font awesome icon

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team#82180

## Testing done

- _Describe what the old behavior was prior to the change_
- _Describe the steps required to verify your changes are working as expected_
- _Describe the tests completed and the results_
- _Exclusively stating 'Specs and automated tests passing' is NOT acceptable as appropriate testing
- _Optionally, provide a link to your [test plan](https://depo-platform-documentation.scrollhelp.site/developer-docs/create-a-test-plan-in-testrail) and [test execution records](https://depo-platform-documentation.scrollhelp.site/developer-docs/execute-tests-in-testrail)_

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| ClaimsListItem.jsx  | Screenshot 2024-05-22 at 11 00 26 AM | Screenshot 2024-05-22 at 12 00 49 PM |
| DueDateOld.jsx | Screenshot 2024-05-22 at 11 57 39 AM | Screenshot 2024-05-22 at 12 28 14 PM |
| SubmittedTrackedItem.jsx | Screenshot 2024-05-22 at 12 32 04 PM | Screenshot 2024-05-22 at 12 37 08 PM |
| DocumnetsFiled.jsx | Screenshot 2024-05-22 at 1 43 21 PM | Screenshot 2024-05-22 at 1 44 43 PM |
| ClaimLetterSection.jsx | Screenshot 2024-05-22 at 2 07 08 PM | Screenshot 2024-05-22 at 2 07 44 PM |
|WhatWeAreDoing.jsx | Screenshot 2024-05-22 at 2 13 59 PM | Screenshot 2024-05-22 at 2 14 30 PM |
|ClaimCardLink.jsx | Screenshot 2024-05-22 at 2 20 48 PM | Screenshot 2024-05-22 at 2 21 19 PM |
| Hover on a chevron link - showing it moves the chevron to the right | Screenshot 2024-05-23 at 9 53 01 AM | Screenshot 2024-05-23 at 9 49 57 AM |
| OnThisPageLink.jsx| Screenshot 2024-05-22 at 2 47 32 PM | Screenshot 2024-05-22 at 3 54 47 PM |

## What areas of the site does it impact?

Claim Status Tool and Rated Disabilities

## Acceptance criteria

- [x] claims-status to no longer uses font awesome icons 
- [x] rated-disabilities to no longer use font awesome icons
- [x] Claim status tool shows the appropriate icons.
- [x] Rated Disabilities shows the appropriate icons.
- [x] Icons use the existing design system

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

### CODE REVIEW

File: src/applications/claims-status/components/ClaimCard/ClaimCardLink.jsx [hunk 1]
```diff
@@ -16,7 +16,7 @@ export default function ClaimCardLink({
       onClick={onClick}
     >
       {text}
-      <i aria-hidden="true" />
+      <va-icon icon="chevron_right" size={3} aria-hidden="true" />
```
Comment (samcoforma) [inline src/applications/claims-status/components/ClaimCard/ClaimCardLink.jsx:19]: I guess my only question so far is regarding the size of these icons. Based on the screenshots, it looks like they were all slightly smaller before?

Comment (pmclaren19) [reply src/applications/claims-status/components/ClaimCard/ClaimCardLink.jsx:19]: Size 3 is the closest that I could get to the original size (where its a little bigger then the text), size 2 makes it smaller then the text and I didnt think we would want that.

Comment (samcoforma) [reply src/applications/claims-status/components/ClaimCard/ClaimCardLink.jsx:19]: Ok, that makes sense

**Review by samcoforma — DISMISSED** [review]

**Comment by jerekshoe** [comment]
For the ClaimLettersSection, WhatWeAreDoing, and the ClaimCardLink, the icon should move to the right when the link is hovered over to match the va-link component. It looks like it's not doing that anymore

**Comment by pmclaren19** [comment]
> For the ClaimLettersSection, WhatWeAreDoing, and the ClaimCardLink, the icon should move to the right when the link is hovered over to match the va-link component. It looks like it's not doing that anymore

Update and added a screen shot of the hover working!

**Review by samcoforma — APPROVED** [review]

**Comment by pmclaren19** [comment]
Verified the new icons with @uxgary good to merge pr!

**Pull request merged**